### PR TITLE
feat: finish svelte5 migration

### PR DESCRIPTION
**Pull request by ndom91**
## ☕️ Reasoning

- Migrate remaining components and files to svelte-5 runes
- Attempt # 2, replacing https://github.com/gitbutlerapp/gitbutler/pull/5395

### CODE REVIEW

**Comment by ndom91** [comment]
Should be good to go. Will test a bit more before merging however.

File: apps/desktop/src/lib/components/AIPromptEdit/AIPromptEdit.svelte [hunk 1]
```diff
@@ -6,30 +6,36 @@
...
+	let prompts = $state<Prompts>();
 
 	if (promptUse === 'commits') {
 		prompts = promptService.commitPrompts;
 	} else {
 		prompts = promptService.branchPrompts;
 	}
 
-	$: userPrompts = prompts.userPrompts;
+	let userPrompts = $derived(prompts.userPrompts);
```
Comment (mtsgrd) [inline apps/desktop/src/lib/components/AIPromptEdit/AIPromptEdit.svelte]: There's too many instances to comment on each one, but let's make derived variables and props `const` where appropriate?

File: apps/desktop/src/lib/components/ProjectSetup.svelte [hunk 2]
```diff
@@ -12,15 +12,19 @@
...
 
-	export let remoteBranches: RemoteBranchInfo[];
+	interface Props {
+		remoteBranches: RemoteBranchInfo[];
+	}
+
+	let { remoteBranches }: Props = $props();
 
-	const project = getContext(Project);
+	const project = $state(getContext(Project));
```
Comment (mtsgrd) [inline apps/desktop/src/lib/components/ProjectSetup.svelte]: This $state looks incorrect, you should be able to just remove it?

File: apps/desktop/src/routes/+page.svelte [hunk 3]
```diff
@@ -2,15 +2,16 @@
 	import { ProjectsService } from '$lib/backend/projects';
 	import FullviewLoading from '$lib/components/FullviewLoading.svelte';
 	import { getContext } from '@gitbutler/shared/context';
-	import { derived } from 'svelte/store';
+	import { run } from 'svelte/legacy';
+	import { derived as derivedStore } from 'svelte/store';
```
Comment (mtsgrd) [inline apps/desktop/src/routes/+page.svelte]: This looks unintentional.

File: apps/desktop/src/routes/+page.svelte [hunk 4]
```diff
@@ -39,13 +40,13 @@
...
 
-	$: {
+	run(() => {
 		if ($redirect.type === 'redirect') {
 			goto($redirect.subject);
 		} else if ($redirect.type === 'no-projects') {
 			goto('/onboarding');
 		}
-	}
+	});
```
Comment (mtsgrd) [inline apps/desktop/src/routes/+page.svelte]: Should we be fixing these `run`s?

File: apps/desktop/src/lib/components/ProjectSetup.svelte (hunk 2 above)
Comment (ndom91) [reply apps/desktop/src/lib/components/ProjectSetup.svelte]: Yeah lemme try that, 90% of this PR was done by the migration tool btw

File: apps/desktop/src/routes/+page.svelte (hunk 3 above)
Comment (ndom91) [reply apps/desktop/src/routes/+page.svelte]: No so the migration tool complained about using the store derived in the same file as runes `$derived`. Let me double check though

File: apps/desktop/src/routes/+page.svelte (hunk 4 above)
Comment (ndom91) [reply apps/desktop/src/routes/+page.svelte]: Yeah slowly but surely going through and removing all the svelte/legacy imports 👍

File: apps/desktop/src/routes/+page.svelte (hunk 3 above)
Comment (mtsgrd) [reply apps/desktop/src/routes/+page.svelte]: ~~That shouldn't be a problem!~~ Right, that makes sense. I was clearly wrong.

**Review by mtsgrd — APPROVED** [review]

File: apps/desktop/src/lib/components/ProjectSetup.svelte (hunk 2 above)
Comment (ndom91) [reply apps/desktop/src/lib/components/ProjectSetup.svelte]: Removed :ok_hand:

**Comment by krlvi** [comment]
shall we land this bad boy?

**Pull request merged**